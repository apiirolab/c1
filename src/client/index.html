<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>