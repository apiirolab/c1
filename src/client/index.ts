export const info = 'Client Info';
