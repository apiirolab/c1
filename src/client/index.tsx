import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
