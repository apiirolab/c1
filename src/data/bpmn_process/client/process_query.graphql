query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
