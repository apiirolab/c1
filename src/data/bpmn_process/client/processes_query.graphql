query Processes($input: BpmnProcessesInput!) {
  processes(input: $input) {
    id
    name
  }
}
