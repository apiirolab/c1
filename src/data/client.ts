import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
