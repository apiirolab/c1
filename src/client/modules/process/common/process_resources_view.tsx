import * as React from 'react';

import { List } from 'semantic-ui-react';

import { Link } from '@reach/router';
import { QueryTypes } from 'data/client';

type Props = {
  context: App.Context;
  process: QueryTypes.BpmnProcessDefinition;
  processInstance: QueryTypes.BpmnProcessInstance;
  filter?: QueryTypes.ResourceType;
};

export class ProcessResources extends React.Component<Props> {
  // renderModal() {
  //   const resource = this.state.activeResource;

  //   return (
  //     <Modal
  //       open={true}
  //       dimmer="blurring"
  //       size="fullscreen"
  //       closeOnDimmerClick={false}
  //       closeOnEscape={false}
  //     >
  //       <Modal.Header>{resource.name}</Modal.Header>
  //       <Modal.Content scrolling>
  //         description
  //         <br />
  //       </Modal.Content>
  //       <Modal.Actions>
  //         <Button
  //           default
  //           content={this.props.context.i18n`Close`}
  //           onClick={() => this.setState({ activeResource: null })}
  //         />
  //       </Modal.Actions>
  //     </Modal>
  //   );
  // }

  createUrl(r: QueryTypes.BpmnProcessResource, id: String) {
    return this.props.processInstance
      ? `/process/${this.props.process.name.url()}/view/${r.type.toLowerCase()}/${r.name.url()}/${
          this.props.processInstance.id
        }/${id}`
      : `/process/${this.props.process.name.url()}/preview/${r.type.toLowerCase()}/${r.name.url()}/${
          this.props.process.id
        }/${id}`;
  }

  render() {
    const { process, filter } = this.props;

    const forms = process.resources.filter(r => r.type === QueryTypes.ResourceType.Form);
    const reports = process.resources.filter(r => r.type === QueryTypes.ResourceType.Report);
    const documents = process.resources.filter(r => r.type === QueryTypes.ResourceType.Document);
    const files = process.resources.filter(r => r.type === QueryTypes.ResourceType.File);
    const links = process.resources.filter(r => r.type === QueryTypes.ResourceType.Url);

    return (
      <>
        {/* {this.state.activeResource && this.renderModal()} */}

        <If condition={!process.resources.length}>
          <I18>ðŸ“‰ No resources allocated to this process</I18>
        </If>
        <List>
          <If condition={!!forms.length && (!filter || filter === QueryTypes.ResourceType.Form)}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Forms</List.Header>
                {/* <List.Description>User forms of this process</List.Description> */}

                <List.List>
                  {forms.map(r => (
                    <List.Item
                      as={Link}
                      to={this.createUrl(r, (r.form || { id: 'error' }).id)}
                      key={r.id}
                      icon="wordpress forms"
                      content={r.name}
                      // onClick={() => this.setState({ activeResource: r })}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If
            condition={!!reports.length && (!filter || filter === QueryTypes.ResourceType.Report)}
          >
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Reports</List.Header>
                <List.List>
                  {reports.map(r => (
                    <List.Item
                      as={Link}
                      to={this.createUrl(r, (r.form || { id: 'error' }).id)}
                      key={r.id}
                      icon="newspaper"
                      content={r.name}
                      onClick={() => this.setState({ activeResource: r })}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If
            condition={
              !!documents.length && (!filter || filter === QueryTypes.ResourceType.Document)
            }
          >
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Documents</List.Header>
                <List.List>
                  {documents.map(r => (
                    <List.Item
                      as={Link}
                      to={this.createUrl(r, r.document.id)}
                      key={r.id}
                      icon="file"
                      content={r.name}
                      onClick={() => this.setState({ activeResource: r })}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If condition={!!files.length && !filter}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Files</List.Header>
                <List.List>
                  {files.map(r => (
                    <List.Item
                      as={Link}
                      target="_blank"
                      key={r.id}
                      icon="file alternate"
                      content={r.name}
                      to={`/files/${process.id}/${r.link}`}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If condition={!!links.length && !filter}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Links</List.Header>
                <List.List>
                  {links.map(r => (
                    <List.Item
                      as="a"
                      target="_blank"
                      key={r.id}
                      icon="linkify"
                      content={r.name}
                      href={r.link}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
        </List>
      </>
    );
  }
}
