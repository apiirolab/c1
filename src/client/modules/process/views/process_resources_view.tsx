import * as React from 'react';

import { Link } from '@reach/router';
import { Header, List, Segment } from 'semantic-ui-react';

import { QueryTypes } from 'data/client';

type Props = {
  context: App.Context;
  process: QueryTypes.BpmnProcessDefinition;
};

export const ProcessResources: React.SFC<Props> = ({ context, process }) => {
  const forms = process.resources.filter(r => r.type === QueryTypes.ResourceType.Form);
  const reports = process.resources.filter(r => r.type === QueryTypes.ResourceType.Report);
  const documents = process.resources.filter(
    r => r.type !== QueryTypes.ResourceType.Form && r.type !== QueryTypes.ResourceType.Report
  );

  return (
    <>
      <Header inverted attached="top" as="h5" icon="file" content="Resources" dividing />
      <Segment attached>
        <If condition={!process.resources.length}>
          <I18>No resources available</I18>
        </If>
        <List>
          <If condition={!!forms.length}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Forms</List.Header>
                {/* <List.Description>User forms of this process</List.Description> */}

                <List.List>
                  {forms.map(r => (
                    <List.Item
                      as={Link}
                      key={r.id}
                      icon="wordpress forms"
                      content={r.name}
                      to={`/forms/${context.Ui.urlName(r.name)}/${r.id}`}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If condition={!!reports.length}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Reports</List.Header>
                <List.List>
                  {reports.map(r => (
                    <List.Item
                      as={Link}
                      key={r.id}
                      icon="newspaper"
                      content={r.name}
                      to={`/reports/${context.Ui.urlName(r.name)}/${r.id}`}
                    />
                  ))}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
          <If condition={!!documents.length}>
            <List.Item>
              <List.Icon name="folder" />
              <List.Content>
                <List.Header as={I18}>Resources</List.Header>
                <List.List>
                  {documents.map(r => {
                    switch (r.type) {
                      case QueryTypes.ResourceType.Document:
                        return (
                          <List.Item
                            as={Link}
                            key={r.id}
                            icon="file"
                            content={r.name}
                            to={`/files/${context.Ui.urlName(r.name)}/${r.id}`}
                          />
                        );
                      case QueryTypes.ResourceType.File:
                        return (
                          <List.Item
                            as={Link}
                            key={r.id}
                            icon="file alternate"
                            content={r.name}
                            to={`/files/${context.Ui.urlName(r.name)}/${r.id}`}
                          />
                        );
                      case QueryTypes.ResourceType.Url:
                        return (
                          <List.Item
                            as="a"
                            key={r.id}
                            icon="linkify"
                            content={r.name}
                            href={r.content}
                          />
                        );
                    }
                  })}
                </List.List>
              </List.Content>
            </List.Item>
          </If>
        </List>
      </Segment>
    </>
  );
};
