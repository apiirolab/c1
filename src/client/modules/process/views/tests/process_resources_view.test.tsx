import * as React from 'react';

import { create, render } from 'client/tests';

import { createProcessInstance } from '../../containers/tests/process_test_data';
import { ProcessResources } from '../process_resources_view';

describe('Process', () => {
  describe('Resources', () => {
    const context = create.context();
    const processInstance = createProcessInstance();

    const component = <ProcessResources process={processInstance.process} context={context} />;

    it('renders correctly', () => {
      expect(render(component)).toMatchSnapshot();
    });

    return { component };
  });
});
