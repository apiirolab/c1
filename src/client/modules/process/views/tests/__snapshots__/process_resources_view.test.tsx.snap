// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Resources renders correctly 1`] = `
Array [
  <h5
    className="ui dividing inverted top attached header"
  >
    <i
      aria-hidden="true"
      className="file icon"
    />
    <div
      className="content"
    >
      Resources
    </div>
  </h5>,
  <div
    className="ui attached segment"
  >
    <div
      className="ui list"
      role="list"
    >
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <i
          aria-hidden="true"
          className="folder icon"
        />
        <div
          className="content"
        >
          Forms
          <div
            className="list"
          >
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <i
                aria-hidden="true"
                className="wordpress forms icon"
              />
              <div
                className="content"
              >
                Approval Form
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <i
          aria-hidden="true"
          className="folder icon"
        />
        <div
          className="content"
        >
          Reports
          <div
            className="list"
          >
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <i
                aria-hidden="true"
                className="newspaper icon"
              />
              <div
                className="content"
              >
                Approval Report
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <i
          aria-hidden="true"
          className="folder icon"
        />
        <div
          className="content"
        >
          Documents
          <div
            className="list"
          >
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <i
                aria-hidden="true"
                className="file icon"
              />
              <div
                className="content"
              >
                Guidelines
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <i
          aria-hidden="true"
          className="folder icon"
        />
        <div
          className="content"
        >
          Files
          <div
            className="list"
          >
            <a
              className="item"
              onClick={[Function]}
              role="listitem"
              target="_blank"
              to="/files/excel-file/3"
            >
              <i
                aria-hidden="true"
                className="file alternate icon"
              />
              <div
                className="content"
              >
                Excel File
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="item"
        onClick={[Function]}
        role="listitem"
      >
        <i
          aria-hidden="true"
          className="folder icon"
        />
        <div
          className="content"
        >
          Links
          <div
            className="list"
          >
            <a
              className="item"
              href="http://google.com"
              onClick={[Function]}
              role="listitem"
              target="_blank"
            >
              <i
                aria-hidden="true"
                className="linkify icon"
              />
              <div
                className="content"
              >
                External Resource
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
